### PR TITLE
I95 setup pwa

### LINKED ISSUE #95: Setup PWA
## Basic Information

- [ ] Use next-pwa

## Other Information

Potential scopes:

### PR DESCRIPTION
**Pull request by skyheat**
## Change Summary
Made the app a progressive web app that allows users to download the app to their devices.

### Change Form

- [x] The pull request title has an issue number
- [x] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [ ] The change has documentation

## Other Information

-  `next.config.js` throws a linting error (no-var-requires) but otherwise works fine.
 ```
const withPWA = require('next-pwa')
```

- App Logo looks kinda bad on iOS. WIll be fixing/using different styles

# Related Issue

- Resolve #95

### CODE REVIEW

File: src/pages/_document.tsx [hunk 1]
```diff
@@ -13,6 +13,9 @@ const Document = () => {
           type='font/woff2'
           crossOrigin='anonymous'
         />
+        <link rel='manifest' href='/manifest.json' />
+        <link rel='apple-touch-icon' href='/icon.png' />
```
Comment (NicholasChoong) [inline src/pages/_document.tsx:17]: should put this in the favicon array

File: public/manifest.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      "type": "image/x-icon",
+      "purpose": "maskable"
+    }
+  ],
+  "theme_color": "#FFFFFF",
+  "background_color": "#FFFFFF",
+  "start_url": "/",
+  "display": "standalone",
+  "orientation": "portrait"
+}
```
Comment (NicholasChoong) [inline public/manifest.json:29]: This is not needed as app.webmanifest already exist. Edge recommends it over manifest.json. 
https://web.dev/learn/pwa/web-app-manifest/

**Review by NicholasChoong — CHANGES_REQUESTED** [review]
looks good

File: public/site.webmanifest [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "",
+    "short_name": "",
```
Comment (NicholasChoong) [inline public/site.webmanifest:3]: need name

**Review by NicholasChoong — DISMISSED** [review]
looks good

**Review by NicholasChoong — APPROVED** [review]
looks good

File: public/site.webmanifest (hunk 3 above)
Comment (skyheat) [reply public/site.webmanifest]: no

**Pull request merged**